train_model.py

```diff
@@ -0,0 +1,89 @@
+import argparse
+import h5py
+import numpy as np
+
+from keras.models import Sequential
+from keras.layers import Flatten, Dense, Lambda, Conv2D, Cropping2D, Dropout, MaxPooling2D
+from keras.optimizers import Adam
+from sklearn.model_selection import train_test_split
+
+from utils.resnet import resnet50_model
+from keras.backend import tensorflow_backend as backend
+
+
+# model name
+NVIDIA = "nvidia"
+RESNET50 = "resnet50"
+
+MODEL_NAME = NVIDIA
+# MODEL_NAME = RESNET50
+
+
+def option_parser() -> str:
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-s', '--source_training_data_path',
+                        required=True,
+                        type=str,
+                        help='path to source training data file')
+    args = parser.parse_args()
+
+    return args.source_training_data_path
+
+
+def trainModelAndSave(model, inputs, outputs, epochs, batch_size):
+    
+    X_train, X_valid, y_train, y_valid = train_test_split(inputs, outputs, test_size=0.2, shuffle=True)
+    #Setting model
+    model.compile(loss='mean_squared_error', optimizer=Adam(lr=1.0e-4))
+    #Learning model
+    model.fit(X_train, y_train, batch_size=batch_size, nb_epoch=epochs, verbose=1, validation_data=(X_valid, y_valid))
+    #Saving model
+    model.save("models/" + MODEL_NAME + ".h5")
+

```

Comment:
[MUST]
リサイズ（画像の前処理）の関数の追加

```suggestion

def preprocessing(model, inputs, outputs):

    # resize to fit model input
    images = []
    for input in inputs:
        image = cv2.resize(input, (model.input_shape[2], model.input_shape[1]), interpolation=cv2.INTER_LINEAR)
        images.append(image)

    return np.array(images), outputs
```